pennylane/ops/op_math/sum.py

```diff
@@ -54,31 +54,6 @@ def op_sum(*summands, do_queue=True, id=None):
     return Sum(*summands, do_queue=do_queue, id=id)
 
 
-def _sum(mats_gen, dtype=None, cast_like=None):
-    r"""Private method to compute the sum of matrices.
-
-    Args:
-        mats_gen (Generator): a python generator which produces the matrices which
-            will be summed together.
-
-    Keyword Args:
-        dtype (str): a string representing the data type of the entries in the result.
-        cast_like (Tensor): a tensor with the desired data type in its entries.
-
-    Returns:
-        res (Tensor): the tensor which is the sum of the matrices obtained from mats_gen.
-    """
-    # Note this method is currently inefficient (improve addition by looking at wire subgroups)
-    res = reduce(math.add, mats_gen)
-
-    if dtype is not None:
-        res = math.cast(res, dtype)
-    if cast_like is not None:
-        res = math.cast_like(res, cast_like)
-
-    return res
```

Comment:
Glad we are removing this, I had added it when I initially implemented the class and it turns out we didn't need the casting functionality at all.